src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1006,17 +1043,25 @@ SinglePartitionJoin(JoinOrderNode *currentJoinNode, TableEntry *candidateTable,
 OpExpr *
 SinglePartitionJoinClause(Var *partitionColumn, List *applicableJoinClauses)
 {
-	ListCell *applicableJoinClauseCell = NULL;
 	if (partitionColumn == NULL)
 	{
 		return NULL;
 	}
 
-	foreach(applicableJoinClauseCell, applicableJoinClauses)
+	Node *applicableJoinClause = NULL;
+	foreach_ptr(applicableJoinClause, applicableJoinClauses)
 	{
-		OpExpr *applicableJoinClause = castNode(OpExpr, lfirst(applicableJoinClauseCell));
-		Var *leftColumn = LeftColumnOrNULL(applicableJoinClause);
-		Var *rightColumn = RightColumnOrNULL(applicableJoinClause);
+		if (!IsA(applicableJoinClause, OpExpr))
+		{
+			continue;
+		}
+		OpExpr *applicableJoinOpExpr = castNode(OpExpr, applicableJoinClause);
+		if (!OperatorImplementsEquality(applicableJoinOpExpr->opno))
+		{
+			continue;
+		}
```

Comment:
This portion is twice in this PR, without any good documentation.

Instead; could we extract this to a function `IsEquiJoinClause`? This makes the code readable as
```C
if (!IsEquiJoinClause(applicableJoinClause))
{
    continue;
}
```